tests/main/views/test_auth.py

```diff
@@ -42,17 +39,15 @@ def test_should_redirect_to_supplier_dashboard_on_supplier_login(self):
         assert res.location == 'http://localhost/suppliers'
         assert 'Secure;' in res.headers['Set-Cookie']
 
-    @mock.patch('app.main.views.auth.data_api_client')
-    def test_should_redirect_to_homepage_on_buyer_login(self, data_api_client):
-        with self.app.app_context():
```

Comment:
that's interesting, why did we need this app context before, but we do not need it anymore? Anything to do with our refactor or just how new versions of our stack work?

Reply:
Ben gave a good explanation of this on the equivalent Briefs FE refactoring PR (https://github.com/alphagov/digitalmarketplace-briefs-frontend/pull/93):

```We shouldn't ever need a self.app.app_context call in the frontend tests. IIRC a call to a frontend view should create a new app and request context for the given request.```

This test is doing a simple `self.client.post(...)` - the app context should be created as part of the client's post request.

We should only occasionally need the `self.app.app_context` inside the test, usually when we want to test how code behaves under a different app config setting. For example in the Buyer FE there's a test where we override `current_app.config['DM_SEARCH_PAGE_SIZE']` - we only want to change it for the scope of that particular test.

Reply:
(I don't know why it was there before - probably a copy and paste from somewhere where we did need it?)

Reply:
Ahhh, ok, thank you 💖